File: leetcode/problems/language-specific/go/190.reverse-bits.go
/*
 * @lc app=leetcode id=190 lang=golang
 *
 * [190] Reverse Bits
 */

// @lc code=start
func reverseBits(num uint32) uint32 {
	res := uint32(0)
	for i := 0; i < 32; i++ {
		res = res<<1 + num&1
		num >>= 1
	}

	return res
}

// @lc code=end

File: leetcode/problems/sessions/2025-10-25_2025-??-??/190.reverse-bits.py
#
# @lc app=leetcode id=190 lang=python3
#
# [190] Reverse Bits
#

# @lc code=start

# * Bit Manipulation Solution | O(1) Time | O(1) Space
# * n -> The given input


class Solution:
    def reverseBits(self, n: int) -> int:
        res = 0
        for _ in range(32):
            res = (res << 1) + (n & 1)
            n >>= 1
        return res


# @lc code=end
